pkg/util/proc/procmanager.go

```diff
@@ -114,7 +116,24 @@ func (p *ProcManager) Shutdown() {
 	p.RUnlock()
 }
 
-func (p *ProcManager) checkProcessExists(binary, procSearchPattern string, policy ProcStartPolicy) (bool, error) {
+// Checks if a process exists. If the restart policy indicates the process should be restarted,
+// the process will be stopped here.
+func (p *ProcManager) checkProcessExists(binary, procSearchPattern string, policy ProcStartPolicy) (shouldStart bool, err error) {
+	// check if this process is already being monitored
+	p.RLock()
+	defer p.RUnlock()
+	if index, proc := p.findMonitoredProc(binary, procSearchPattern); proc != nil {
+		if policy == ReuseExisting {
+			logger.Infof("Process not running, but will be started by the process monitor. %s", procSearchPattern)
```

Comment:
I'll clarify the comments around this. If we find that the process is already being monitored and policy is ReuseExisting, this means that we want the process manager to worry about managing the process. It will ensure that the process is started. You're correct that the log statement is incorrect because the process might be running or might not be, we don't know here. We only know it's under management.

We return false when the user is supposed to start the process and add it under management. But there is no need to add it to be under management if it's already managed.